src/ckb_chain/contracts.rs

```diff
@@ -394,12 +392,35 @@ impl ContractsContext {
             Self::Real(real) => &real.contract_default_scripts,
         }
     }
-    pub(crate) fn get_cell_deps(&self, _contracts: Vec<Contract>) -> CellDepVec {
-        match self {
+    pub(crate) fn get_cell_deps(&self, contracts: Vec<Contract>) -> CellDepVec {
+        let (script_cell_deps, contracts) = match self {
             #[cfg(test)]
-            Self::Mock(mock) => mock.contracts_context.cell_deps.clone(),
-            Self::Real(real) => real.cell_deps.clone(),
+            Self::Mock(mock) => {
+                //FIXME(yukang): why we need this
+                let mut contracts = contracts;
+                contracts.push(Contract::CkbAuth);
+                (&mock.contracts_context.script_cell_deps, contracts)
+            }
+            Self::Real(real) => (&real.script_cell_deps, contracts),
+        };
+
+        let cell_deps_vec = contracts
+            .into_iter()
+            .map(|contract| {
+                script_cell_deps
+                    .get(&contract)
```

Comment:
We can't simple find cell deps by lock up the hash map, as some contracts actually require a few more dependencies. E.g. funding lock and commitment lock both require ckb auth binary. Commitment lock may additionally require the sudt binary.

Reply:
Currently, in dev net, funding lock and commitment lock is already configured as a dep group. https://github.com/nervosnetwork/cfn-node/blob/main/tests/deploy/migrations/templates/funding-lock.toml . We need to make testnet and mainnet contracts as dep groups.